data_processing/cleaning_demographic_data.Rmd

```diff
@@ -52,11 +50,412 @@ demo_dat %>%
+  filter(str_detect(aps_report_location, "-")) %>% 
   filter(str_detect(aps_report_location, "-"))
 ```
 
+replace all locations that contain Roseville in the name to just Roseville as a new location variable 
+```{r}
+demo_dat <- demo_dat %>% 
+  mutate(aps_location_recode = case_when(str_detect(aps_report_location,"rosev") ~ "roseville",
+                                      
+                                         TRUE ~ aps_report_location))
+```
+
+check
+```{r}
+demo_dat %>% 
+  count(aps_report_location, aps_location_recode) %>% 
+  filter(str_starts(aps_report_location, "r"))
+```
+Location of Participation:
+remove "'",", ca" and "ca" from the end of responses
+```{r}
+demo_dat <- demo_dat %>% 
+  mutate(location_of_participation_recode = str_replace(location_of_participation, "'", "")) %>% 
+  mutate(location_of_participation_recode = str_replace(location_of_participation, ", ca| ca|,ca", ""))
+```
+
+Check for removal of Ca
+```{r}
+demo_dat %>% 
+  filter(str_detect(location_of_participation, ",")) %>% 
+  count(location_of_participation, location_of_participation_recode)
+```
+
+remove special characters like ":" before laguna beach. Combine misspellings of bakersfield.Combine mispelling of barstow.
+```{r}
+demo_dat <- demo_dat %>% 
+  mutate(location_of_participation_recode = case_when(str_detect(location_of_participation, ":") ~ "laguna beach",
+                                                      str_detect(location_of_participation, "bakersf|bakerf") ~ "bakersfield",
+                                                      str_detect(location_of_participation, "bartso") ~ "barstow",
+                                                      TRUE ~ location_of_participation))
+```
+
+
+Race_1 - replace american indian/alaskan native with american indian/alaskan native/indigenous. Replace asian with asian/asian american. Replace black/african american with black/african american/african. Replace pacific islander/native hawaiian with native hawaiian/pacific islander. Replace whte with white. Change all unknowns, including data not collected and n/a to "unknown".
+```{r}
+demo_dat <- demo_dat %>% 
+  mutate(race_1_recode = case_when(str_detect(race_1,"americ") ~ "american indian/alaskan native/indigenous", 
+                                      str_starts(race_1, "asian") ~ "asian/asian american",
+                                      str_starts(race_1, "bla") ~ "black/african american/african",
+                                      str_starts(race_1, "pacific i") ~ "native hawaiian/pacific islander",
+                                      str_starts(race_1, "wh") ~ "white",
+                                   str_detect(race_1, "unkn|#|client|collected") ~ "unknown",
+                                   is.na(race_1) ~ "unknown",
+                                      
+                                         TRUE ~ race_1))
+```
+
+check
+```{r}
+demo_dat %>% 
+  count(race_1, race_1_recode)
+```
+
```

Comment:
In the current md file that was rendered it looks like there are a bunch of observations that are a number string that starts with 4, I don't know if this gets fixed or not in the recent data dump, but these should be coded to `unknown` for example,
`str_starts(race_1, "4") ~ "unknown"`